src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs

```diff
@@ -388,21 +505,432 @@ public void RegisterMessageHandler(Func<Message, CancellationToken, Task> handle
             this.OnMessageHandlerAsync(messageHandlerOptions, handler).GetAwaiter().GetResult();
         }
 
-        protected abstract Task<IList<Message>> OnReceiveAsync(int maxMessageCount, TimeSpan serverWaitTime);
+        public override async Task OnClosingAsync()
```

Comment:
>OnClosingAsync [](start = 35, length = 14)

not completely related to your change, but could you please make this `OnClosingAsync()` protected EVERYWHERE?